react/features/welcome/components/WelcomePage.web.js

```diff
@@ -1,11 +1,112 @@
-import React, { Component } from 'react';
+/* global interfaceConfig, APP  */
+
+import React from 'react';
+import { connect } from 'react-redux';
+
+import {
+    AbstractWelcomePage,
+    mapStateToProps
+} from './AbstractWelcomePage';
+
+const RIGHT_WATERMARK_STYLES = {
+    backgroundImage: 'url(images/rightwatermark.png)'
+};
 
 import { Conference } from '../../conference';
 
 /**
  * The web container rendering the welcome page.
  */
-export default class WelcomePage extends Component {
+class WelcomePage extends AbstractWelcomePage {
+
+    /**
+    * Constructor function of WelcomePage.
+    *
+    * @param {Object} props - Props to be set.
+    **/
+    constructor(props) {
+        super(props);
+        this._initState();
+
+        // Bind event handlers so they are only bound once for every instance.
+        const onToggleDisableWelcome = this._onToggleDisableWelcomePage;
+
+        this._onRoomChange = this._onRoomChange.bind(this);
+        this._onKeyDown = this._onKeyDown.bind(this);
+        this._setInput = this._setInput.bind(this);
+        this._onUpdateRoomname = this._onUpdateRoomname.bind(this);
+        this._onToggleDisableWelcomePage = onToggleDisableWelcome.bind(this);
+    }
+
+    /**
+    * Method that initializes state of the component.
+    *
+    * @returns {void}
+    **/
+    _initState() {
+        const showPoweredBy = interfaceConfig.SHOW_POWERED_BY;
+        const generateRoomnames
+            = interfaceConfig.GENERATE_ROOMNAMES_ON_WELCOME_PAGE;
+        const enableWelcomePage = true;
+        const showJitsiWatermark = interfaceConfig.SHOW_JITSI_WATERMARK;
+        const showBrandWatermark = interfaceConfig.SHOW_BRAND_WATERMARK;
+        let jitsiWatermarkLink = '';
+        let brandWatermarkLink = '';
+
+        if (showJitsiWatermark) {
+            jitsiWatermarkLink = interfaceConfig.JITSI_WATERMARK_LINK;
+        }
+
+        if (showBrandWatermark) {
+            brandWatermarkLink = interfaceConfig.BRAND_WATERMARK_LINK;
+        }
+
+        this.state = Object.assign({}, this.state, {
+            showPoweredBy,
+            generateRoomnames,
+            showJitsiWatermark,
+            showBrandWatermark,
+            jitsiWatermarkLink,
+            brandWatermarkLink,
+            enableWelcomePage
+        });
+    }
+
+    /**
+    * Returns the domain name.
+    *
+    * @private
+    * @returns {string} Domain name.
+    **/
+    _getDomain() {
+        return `${window.location.protocol}//${window.location.host}/`;
+    }
+
+    /**
+    * This method is executed when comonent is mounted.
+    *
+    * @inheritdoc
+    */
```

Comment:
1. That's not the way we format jsdoc comments.
2. You've introduced 2 different non-compliant jsdoc comment formats in 1 file.